### PR TITLE
Validate regex before building image for e2e test

### LINKED ISSUE #4750: PR e2e test: Validate the regex before building the images
### Proposal

Currently, we build the images and deploy them into the cluster before checking the provided regex, this could deal in a failure 20 minutes later. 
We should check somehow the regex before starting the process to be more efficient and saving resources

### Use-Case

_No response_

### Is this a feature you are interested in implementing yourself?

Yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Yaxhveer**
Added the Regex check before building image and made refactored changes accordingly.

### Checklist

- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #4750 

Relates to #

### CODE REVIEW

**Comment by Yaxhveer** [comment]
@JorTurFer PTAL.

File: .github/workflows/pr-e2e.yml [hunk 1]
```diff
@@ -157,14 +168,7 @@ jobs:
...
           E2E_IMAGE_TAG: ${{ needs.triage.outputs.image_tag }}
           TEST_CLUSTER_NAME: keda-e2e-cluster-pr
-          COMMENT_BODY: ${{ github.event.comment.body }}
         run: |
-          MESSAGE="$COMMENT_BODY"
-          REGEX='/run-e2e (.+)'
-          if [[ "$MESSAGE" =~ $REGEX ]]
-          then
-            export E2E_TEST_REGEX="$(echo ${BASH_REMATCH[1]} | head -1)"
-          fi
```
Comment (JorTurFer) [inline .github/workflows/pr-e2e.yml:167]: I think that we still need this code for filtering the regex during the e2e test execution

File: .github/workflows/pr-e2e.yml [hunk 2]
```diff
@@ -93,6 +93,17 @@ jobs:
...
+      - name: Run regex checks
+        env:
+          COMMENT_BODY: ${{ github.event.comment.body }}
+        run: |
+          MESSAGE="$COMMENT_BODY"
+          REGEX='/run-e2e (.+)'
+          if [[ "$MESSAGE" =~ $REGEX ]]
+          then
+            export E2E_TEST_REGEX="$(echo ${BASH_REMATCH[1]} | head -1)"
+          fi
```
Comment (JorTurFer) [inline .github/workflows/pr-e2e.yml:105]: We need to call to `e2e-regex-check`
```suggestion
      - name: Run regex checks
        id: regex-validation
        env:
          COMMENT_BODY: ${{ github.event.comment.body }}
        run: |
          MESSAGE="$COMMENT_BODY"
          REGEX='/run-e2e (.+)'
          if [[ "$MESSAGE" =~ $REGEX ]]
          then
            export E2E_TEST_REGEX="$(echo ${BASH_REMATCH[1]} | head -1)"
          fi
          make e2e-regex-check
```

File: .github/workflows/pr-e2e.yml [hunk 3]
```diff
@@ -93,6 +93,17 @@ jobs:
...
+
```
Comment (JorTurFer) [inline .github/workflows/pr-e2e.yml:106]: We need to include there the failure of the check and the feedback:
```suggestion

      - name: React to comment with failure
        uses: dkershner6/reaction-action@v2
        if: steps.regex-validation.outcome != 'success'
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          commentId: ${{ github.event.comment.id }}
          reaction: "-1"

      - name: Set status failure
        uses: LouisBrunner/checks-action@6b626ffbad7cc56fd58627f774b9067e6118af23 # v2
        if: steps.regex-validation.outcome != 'success'
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          sha: ${{ needs.triage.outputs.commit_sha }}
          name: ${{ env.E2E_CHECK_NAME }}
          conclusion: failure
          details_url: https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}

```

**Comment by JorTurFer** [comment]
> /run-e2e nonsense **Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/9125967954)

This won't help here, because the workflow is executed from `main` branch because it's a comment flow

File: .github/workflows/pr-e2e.yml (hunk 2 above)
Comment (Yaxhveer) [reply .github/workflows/pr-e2e.yml]: Yes, forgot to add that.

File: .github/workflows/pr-e2e.yml [hunk 4]
```diff
@@ -93,6 +93,41 @@ jobs:
...
+        env:
+          COMMENT_BODY: ${{ github.event.comment.body }}
+        run: |
+          MESSAGE="$COMMENT_BODY"
+          REGEX='/run-e2e (.+)'
+          if [[ "$MESSAGE" =~ $REGEX ]]
+          then
+            export E2E_TEST_REGEX="$(echo ${BASH_REMATCH[1]} | head -1)"
+          fi
+          make e2e-regex-check
```
Comment (JorTurFer) [inline .github/workflows/pr-e2e.yml:107]: Only the id is missing :)
```suggestion
      - name: Run regex checks
        id: regex-validation
        continue-on-error: true
        env:
          COMMENT_BODY: ${{ github.event.comment.body }}
        run: |
          MESSAGE="$COMMENT_BODY"
          REGEX='/run-e2e (.+)'
          if [[ "$MESSAGE" =~ $REGEX ]]
          then
            export E2E_TEST_REGEX="$(echo ${BASH_REMATCH[1]} | head -1)"
          fi
          make e2e-regex-check
```

**Review by JorTurFer — COMMENTED** [review]
Just

**Review by JorTurFer — APPROVED** [review]
LGTM! Let's merge this and check if it works (sadly, we cannot test the change here during the PR 🤷 )

**Pull request merged**